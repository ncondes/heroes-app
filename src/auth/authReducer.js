import { types } from '../types/types';

// const state = {
//     name: 'Nicolas',
//     logged: true
// }

// const loginAction = {
//     type: types.login,
//     payload: {
//         name: 'Nicolas',
//         email: 'Nicolas@gmail.com'
//     }
// }

export const authReducer = ( state = {}, action ) => {

    switch ( action.type ) {
        case types.login:
            return {
                ...action.payload,
                logged: true,
            }
        case types.logout:
            return {
                logged: false,
            }
    
        default:
            return state;
    }

}
