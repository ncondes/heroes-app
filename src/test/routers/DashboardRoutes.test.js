import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../auth/authContext';
import { DashboardRoutes } from '../../routers/DashboardRoutes';


describe('Pruebas en DashboardRoutes', () => {

    const contextValue = {
        user: {
            name: 'Ricardo',
            logged: true,
        }
    };

    test('Deberia de mostrarse correctamente la ruta de marvel', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                {/* Memory Router ayuda a proveer el router necesario para hacer pruebas, ya que el useNavigate que se usa en algunas de las rutas requiere estar dentro de un router */}
                <MemoryRouter initialEntries={ ['/'] }>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.text-info').text().trim() ).toBe( 'Ricardo' );

    });

    test('Deberia de mostrarse correctamente la ruta de dc', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                {/* Memory Router ayuda a proveer el router necesario para hacer pruebas, ya que el useNavigate que se usa en algunas de las rutas requiere estar dentro de un router */}
                <MemoryRouter initialEntries={ ['/dc'] }>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h1').text().trim() ).toBe( 'DC' )

    });
    
})