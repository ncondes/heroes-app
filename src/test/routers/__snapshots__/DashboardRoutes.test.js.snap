// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en DashboardRoutes Deberia de mostrarse correctamente la ruta de dc 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="navbar-collapse collapse justify-content-between"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            className="nav-link false"
            href="/marvel"
            onClick={[Function]}
          >
            Marvel
          </a>
          <a
            aria-current="page"
            className="nav-link active"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            className="nav-link false"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </div>
        <div
          className="navbar-nav"
        >
          <span
            className="nav-link text-info"
          >
            Ricardo
          </span>
          <button
            className="nav-link btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        DC
      </h1>
      <hr />
      <div
        className="row rows-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/dc-batman"
          onClick={[Function]}
        >
          <img
            alt="Batman"
            className="img img-responsive"
            src="./assets/dc-batman.jpg"
          />
          <div
            className="profile-name"
          >
            Batman
          </div>
          <div
            className="profile-position"
          >
            Bruce Wayne
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Detective Comics #27
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/dc-superman"
          onClick={[Function]}
        >
          <img
            alt="Superman"
            className="img img-responsive"
            src="./assets/dc-superman.jpg"
          />
          <div
            className="profile-name"
          >
            Superman
          </div>
          <div
            className="profile-position"
          >
            Kal-El
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Action Comics #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/dc-flash"
          onClick={[Function]}
        >
          <img
            alt="Flash"
            className="img img-responsive"
            src="./assets/dc-flash.jpg"
          />
          <div
            className="profile-name"
          >
            Flash
          </div>
          <div
            className="profile-position"
          >
            Jay Garrick
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Flash Comics #1
                  </p>
                  <p>
                    Jay Garrick, Barry Allen, Wally West, Bart Allen
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/dc-green"
          onClick={[Function]}
        >
          <img
            alt="Green Lantern"
            className="img img-responsive"
            src="./assets/dc-green.jpg"
          />
          <div
            className="profile-name"
          >
            Green Lantern
          </div>
          <div
            className="profile-position"
          >
            Alan Scott
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    All-American Comics #16
                  </p>
                  <p>
                    Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/dc-arrow"
          onClick={[Function]}
        >
          <img
            alt="Green Arrow"
            className="img img-responsive"
            src="./assets/dc-arrow.jpg"
          />
          <div
            className="profile-name"
          >
            Green Arrow
          </div>
          <div
            className="profile-position"
          >
            Oliver Queen
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    More Fun Comics #73
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/dc-wonder"
          onClick={[Function]}
        >
          <img
            alt="Wonder Woman"
            className="img img-responsive"
            src="./assets/dc-wonder.jpg"
          />
          <div
            className="profile-name"
          >
            Wonder Woman
          </div>
          <div
            className="profile-position"
          >
            Princess Diana
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    All Star Comics #8
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/dc-martian"
          onClick={[Function]}
        >
          <img
            alt="Martian Manhunter"
            className="img img-responsive"
            src="./assets/dc-martian.jpg"
          />
          <div
            className="profile-name"
          >
            Martian Manhunter
          </div>
          <div
            className="profile-position"
          >
            J'onn J'onzz
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Detective Comics #225
                  </p>
                  <p>
                    Martian Manhunter
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/dc-robin"
          onClick={[Function]}
        >
          <img
            alt="Robin/Nightwing"
            className="img img-responsive"
            src="./assets/dc-robin.jpg"
          />
          <div
            className="profile-name"
          >
            Robin/Nightwing
          </div>
          <div
            className="profile-position"
          >
            Dick Grayson
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Detective Comics #38
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/dc-blue"
          onClick={[Function]}
        >
          <img
            alt="Blue Beetle"
            className="img img-responsive"
            src="./assets/dc-blue.jpg"
          />
          <div
            className="profile-name"
          >
            Blue Beetle
          </div>
          <div
            className="profile-position"
          >
            Dan Garret
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Mystery Men Comics #1
                  </p>
                  <p>
                    Dan Garret, Ted Kord, Jaime Reyes
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/dc-black"
          onClick={[Function]}
        >
          <img
            alt="Black Canary"
            className="img img-responsive"
            src="./assets/dc-black.jpg"
          />
          <div
            className="profile-name"
          >
            Black Canary
          </div>
          <div
            className="profile-position"
          >
            Dinah Drake
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    DC Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Flash Comics #86
                  </p>
                  <p>
                    Dinah Drake, Dinah Lance
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas en DashboardRoutes Deberia de mostrarse correctamente la ruta de marvel 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="navbar-collapse collapse justify-content-between"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            className="nav-link false"
            href="/marvel"
            onClick={[Function]}
          >
            Marvel
          </a>
          <a
            className="nav-link false"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            className="nav-link false"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
        </div>
        <div
          className="navbar-nav"
        >
          <span
            className="nav-link text-info"
          >
            Ricardo
          </span>
          <button
            className="nav-link btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Marvel
      </h1>
      <hr />
      <div
        className="row rows-cols-1 row-cols-md-3 g-3 animate__animated animate__fadeIn"
      >
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-spider"
          onClick={[Function]}
        >
          <img
            alt="Spider Man"
            className="img img-responsive"
            src="./assets/marvel-spider.jpg"
          />
          <div
            className="profile-name"
          >
            Spider Man
          </div>
          <div
            className="profile-position"
          >
            Peter Parker
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Amazing Fantasy #15
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-captain"
          onClick={[Function]}
        >
          <img
            alt="Captain America"
            className="img img-responsive"
            src="./assets/marvel-captain.jpg"
          />
          <div
            className="profile-name"
          >
            Captain America
          </div>
          <div
            className="profile-position"
          >
            Steve Rogers
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Captain America Comics #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-iron"
          onClick={[Function]}
        >
          <img
            alt="Iron Man"
            className="img img-responsive"
            src="./assets/marvel-iron.jpg"
          />
          <div
            className="profile-name"
          >
            Iron Man
          </div>
          <div
            className="profile-position"
          >
            Tony Stark
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Tales of Suspense #39
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-thor"
          onClick={[Function]}
        >
          <img
            alt="Thor"
            className="img img-responsive"
            src="./assets/marvel-thor.jpg"
          />
          <div
            className="profile-name"
          >
            Thor
          </div>
          <div
            className="profile-position"
          >
            Thor Odinson
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Journey into Myster #83
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-hulk"
          onClick={[Function]}
        >
          <img
            alt="Hulk"
            className="img img-responsive"
            src="./assets/marvel-hulk.jpg"
          />
          <div
            className="profile-name"
          >
            Hulk
          </div>
          <div
            className="profile-position"
          >
            Bruce Banner
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    The Incredible Hulk #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-wolverine"
          onClick={[Function]}
        >
          <img
            alt="Wolverine"
            className="img img-responsive"
            src="./assets/marvel-wolverine.jpg"
          />
          <div
            className="profile-name"
          >
            Wolverine
          </div>
          <div
            className="profile-position"
          >
            James Howlett
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    The Incredible Hulk #180
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-daredevil"
          onClick={[Function]}
        >
          <img
            alt="Daredevil"
            className="img img-responsive"
            src="./assets/marvel-daredevil.jpg"
          />
          <div
            className="profile-name"
          >
            Daredevil
          </div>
          <div
            className="profile-position"
          >
            Matthew Michael Murdock
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Daredevil #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-hawkeye"
          onClick={[Function]}
        >
          <img
            alt="Hawkeye"
            className="img img-responsive"
            src="./assets/marvel-hawkeye.jpg"
          />
          <div
            className="profile-name"
          >
            Hawkeye
          </div>
          <div
            className="profile-position"
          >
            Clinton Francis Barton
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    Tales of Suspense #57
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-cyclops"
          onClick={[Function]}
        >
          <img
            alt="Cyclops"
            className="img img-responsive"
            src="./assets/marvel-cyclops.jpg"
          />
          <div
            className="profile-name"
          >
            Cyclops
          </div>
          <div
            className="profile-position"
          >
            Scott Summers
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    X-Men #1
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          className="my-card animate__animated animate__fadeIn"
          href="/hero/marvel-silver"
          onClick={[Function]}
        >
          <img
            alt="Silver Surfer"
            className="img img-responsive"
            src="./assets/marvel-silver.jpg"
          />
          <div
            className="profile-name"
          >
            Silver Surfer
          </div>
          <div
            className="profile-position"
          >
            Norrin Radd
          </div>
          <div
            className="profile-overview"
          >
            <div
              className="profile-overview"
            >
              <div
                className="row"
              >
                <div
                  className="col-ms-4"
                >
                  <h3>
                    Marvel Comics
                  </h3>
                  <p>
                    Primera aparición: 
                    <br />
                    The Fantastic Four #48
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>,
]
`;
